### PR TITLE
fix: CalDAV query with time range not being an array

### LINKED ISSUE #39922: Trying to access array offset on value of type bool at CalDavBackend.php#996
### Steps to reproduce
1. Outlook 2019 + CalDav Synchronizer. Also mobile phone Android + DavX two way synchronization enabled.
2. Create calendar and enable two way integration on both devices. It usually works fine but sometimes starts logging this error every automatic or forced sync.
3. We believe request fails when Outlook's CalDav start querying time ranges.

### Expected behaviour
No error in log at each sync request.

### Actual behaviour
Error in log: Trying to access array offset on value of type bool at apps\\dav\\lib\\CalDAV\\CalDavBackend.php#996

### Our solution
We did additional "isset" check on $timeRange variable (seems it is a boolean sometimes) and issue was gone.

Source CalDavBackend.php#996:
if (!$filters['prop-filters'] && !$filters['comp-filters'][0]['comp-filters'] && !$filters['comp-filters'][0]['prop-filters'] && (!$timeRange['start'] || !$timeRange['end']))

Fixed CalDavBackend.php#996:
if (!$filters['prop-filters'] && !$filters['comp-filters'][0]['comp-filters'] && !$filters['comp-filters'][0]['prop-filters'] && (!isset($timeRange['start']) || !isset($timeRange['end'])))

### Server configuration
**Operating system**:
Windows Server 2016

**Web server:**
IIS10

**Database:**
10.6.4-MariaDB, for Win64 (AMD64)

**PHP version:**
PHP 7.4.27 (cli) (built: Dec 14 2021 19:52:04) ( NTS Visual C++ 2017 x64 )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with Zend OPcache v7.4.27, Copyright (c), by Zend Technologies

**ownCloud version:** (see ownCloud admin page)
10.9.1

**Updated from an older ownCloud or fresh install:**
Updated two times before + db upgraded

**Where did you install ownCloud from:**
ZIP package

**Signing status (ownCloud 9.0 and above):**
Signing fails because of our modifications to make ownCloud working on Windows platform. But reported php file with error was untouched. BTW. If you are interested in working Windows version we are open to share it. We are successfully using it since over a year :). 

**The content of config/config.php:**
"config": {
        "dbtype": "mysql",
        "version": "10.9.1.2",
        "dbname": "owncloud",
        "dbhost": "localhost:3306",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "installed": true,
        "maintenance": false,
        "loglevel": 3,
        "log_rotate_size": 10485760,
        "integrity.check.disabled": true,
        "filelocking.enabled": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\APCu",
    }

**List of activated apps:**
Enabled:
  - announcementcenter:
    - Version: 1.2.2
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/announcementcenter
  - calendar:
    - Version: 1.6.4
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps-external/calendar
  - comments:
    - Version: 0.3.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/comments
  - configreport:
    - Version: 0.2.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/configreport
  - contacts:
    - Version: 1.5.5
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps-external/contacts
  - dav:
    - Version: 0.7.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/dav
  - federatedfilesharing:
    - Version: 0.5.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/federatedfilesharing
  - federation:
    - Version: 0.1.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/federation
  - files:
    - Version: 1.5.2
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/files
  - files_external:
    - Version: 0.8.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/files_external
  - files_sharing:
    - Version: 0.14.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/files_sharing
  - files_trashbin:
    - Version: 0.9.1
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/files_trashbin
  - files_versions:
    - Version: 1.3.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/files_versions
  - firstrunwizard:
    - Version: 1.2.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/firstrunwizard
  - market:
    - Version: 0.6.3
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps-external/market
  - notifications:
    - Version: 0.5.4
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/notifications
  - oauth2:
    - Version: 0.5.2
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps-external/oauth2
  - tasks:
    - Version: 0.9.7
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps-external/tasks
  - updatenotification:
    - Version: 0.2.1
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/updatenotification
  - user_ldap:
    - Version: 0.16.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps/user_ldap
  - web:
    - Version: 5.2.0
    - Path: C:/inetpub/www/cloud.app-solutions.pl/apps-external/web

**Are you using external storage, if yes which one:**
no

**Are you using encryption:**
no

**Are you using an external user-backend, if yes which one:**
no

### Client configuration
**Browser:**
N/A

**Operating system:**
Windows 10

#### ownCloud log (data/owncloud.log)
{"reqId":"N0UL6cg9u1qhRr4KcRRl","level":3,"time":"2022-02-19T16:46:49+00:00","remoteAddr":"192.168.0.2","user":"xxx","app":"PHP","method":"REPORT","url":"\/remote.php\/dav\/calendars\/xxx\/calendar\/","message":"Trying to access array offset on value of type bool at xxx\\apps\\dav\\lib\\CalDAV\\CalDavBackend.php#996"}

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Related Issue
- Fixes https://github.com/owncloud/core/issues/39922

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: apps/dav/tests/unit/CalDAV/CalDavBackendTest.php [hunk 1]
```diff
@@ -402,6 +402,7 @@ public function providesCalendarQueryParameters() {
 			'all' => [[0, 1, 2, 3], [], []],
 			'only-todos' => [[], ['name' => 'VTODO'], []],
 			'only-events' => [[0, 1, 2, 3], [], [['name' => 'VEVENT', 'is-not-defined' => false, 'comp-filters' => [], 'time-range' => ['start' => null, 'end' => null], 'prop-filters' => []]],],
+			'timernage false' => [[0, 1, 2, 3], [], [['name' => 'VEVENT', 'is-not-defined' => false, 'comp-filters' => [], 'time-range' => false, 'prop-filters' => []]],],
```
Comment (phil-davis) [inline apps/dav/tests/unit/CalDAV/CalDavBackendTest.php:405]:
```suggestion
			'timerange false' => [[0, 1, 2, 3], [], [['name' => 'VEVENT', 'is-not-defined' => false, 'comp-filters' => [], 'time-range' => false, 'prop-filters' => []]],],
```

Typo in the key - actually it makes no difference to the test data and result.

**Review by jvillafanez — APPROVED** [review]
other than the typo, the code looks good

**Comment by rainforest1155** [comment]
@DeepDiver1975 Is it just me or did the typo @jvillafanez seems to have pointed out is merged? 
I think it’s line 405 where I see:
`'timernage false' => [[0, 1, 2, 3], [], [['name' => 'VEVENT', 'is-not-defined' => false, 'comp-filters' => [], 'time-range' => false, 'prop-filters' => []]],], `

`timernage` looks very much like a typo to me.

**Comment by DeepDiver1975** [comment]
Indeed something got mixed up here..... Damn ....

**Pull request merged**